examples/viewer.py

```diff
@@ -263,15 +286,28 @@ def draw_event(
 
         keys = active_agent_id_and_sensor_name
 
-        self.sim._Simulator__sensors[keys[0]][keys[1]].draw_observation()
-        agent = self.sim.get_agent(keys[0])
-        self.render_camera = agent.scene_node.node_sensor_suite.get(keys[1])
-        self.debug_draw()
-        self.render_camera.render_target.blit_rgba_to_default()
-        mn.gl.default_framebuffer.bind()
+        if self.enable_batch_renderer:
+            for i in range(self.num_env):
+                # Apply keyframe
+                keyframe = self.sim[i].gfx_replay_manager.extract_keyframe()
```

Comment:
This logic here with sensors/set_sensor_transform/etc could be encapsulated in a separate function to keep this draw function a bit more readable.